seacrowd/sea_datasets/mkqa/mkqa.py

```diff
@@ -0,0 +1,245 @@
+                            }
+                        ]
+                        for cur_lang in ([lang] if lang else self._SOURCE_LANGUAGES)
+                    },
+                    "queries": {cur_lang: datasets.Value("string") for cur_lang in ([lang] if lang else self._SOURCE_LANGUAGES)},
+                    "example_id": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
+            features["meta"]["answer_entity"] = datasets.Sequence(datasets.Value("string"))
+            features["meta"]["answer_aliases"] = datasets.Sequence(datasets.Sequence(datasets.Value("string")))
+
+        else:  # schema not found! should NOT reach here ...
+            raise NotImplementedError()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_path = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_path},
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        lang = self.config.subset_id.rsplit("_", 1)[-1]
+        lang = self._LANG_3TO2.get(lang, lang)
+
+        datas = []
+        with open(filepath, "r", encoding="utf8") as ipt:
+            for cur in map(json.loads, ipt):
+                cur["example_id"] = str(cur["example_id"])
+                for key in ["answers", "queries"]:
+                    cur[key] = {k: v for k, v in cur[key].items() if k in ([lang] if lang else self._SOURCE_LANGUAGES)}
+                datas.append(cur)
+
+        if self.config.schema == "source":
+            for cur in datas:
+                for anslist in cur["answers"].values():
+                    for ans in anslist:
+                        ans.setdefault("entity", "")
+                        ans.setdefault("aliases", [])
+                yield int(cur["example_id"]), cur
+
+        elif self.config.schema == "seacrowd_qa":
+            for cur in datas:
+                for cur_lang in [lang] if lang else map(lambda k: self._LANG_3TO2.get(k, k), _LANGUAGES):
+                    ret = {
+                        "id": f'{cur["example_id"]}_{cur_lang}',
+                        "question_id": cur["example_id"],
+                        "document_id": "",
+                        "question": cur["queries"][cur_lang],
+                        "type": [ans["type"] for ans in cur["answers"][cur_lang]],
```

Comment:
`"type": "multiple_choice"`

Could we move the current answer type inside the `"meta"` dictionary?

Reply:
@holylovenia 

> `"type": "multiple_choice"`

Where did you get this value from? There is **no** such value in the original data, CMIIW.
Please see the [source schema for all possible values](https://github.com/SEACrowd/seacrowd-datahub/pull/177/files#diff-b487ae7afd1929c60e38abb6fcf6b13763043e2fb8b04ebd446d49eeaf927aceR158-R169).

---
> Could we move the current answer type inside the `"meta"` dictionary?

Won't that make it coarser-grained?
Following the [Schema Documentation guideline](https://github.com/SEACrowd/seacrowd-datahub/blob/master/task_schemas.md#question-answering), the most appropriate mapping for the field `type` in the original data to the Question Answering Schema is the field `type`. CMIIW.

Do note that `"type": "yesno"` as  written in the [guideline](https://github.com/SEACrowd/seacrowd-datahub/blob/master/task_schemas.md#question-answering) is equivalent to `"type": "binary"` here.
In fact, I have been wondering why the field `type` of QA Schema does not use the more fine-grained alternative (e.q.: `datasets.ClassLabel`) but rather uses a free-text form.
Elaborated explanation on this would be appreciated. 🙏

Reply:
Hey @fhudi, sorry our guide was not up-to-date. The `type` variable is commonly used to show what type of QA it is. For example: [QASiNa](https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/sea_datasets/qasina/qasina.py) is `extractive` and [XCOPA](https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/sea_datasets/xcopa/xcopa.py) is `multiple_choice`.

Reply:
@holylovenia this dataset has no `multiple_choice` though ... 👀
and the `type` variable **_is_** used to show what type of QA it is ...

Reply:
Oh yeah, my bad. I just realized there's no multiple choice... And this dataset also has no context, so maybe we can use `"type": "open_domain"` and move the current answer type inside the "meta" dictionary.

What do you think? @fhudi

Reply:
Ah I see. sounds good. will upade ASAP. @holylovenia

Reply:
> Ah I see. sounds good. will upade ASAP. @holylovenia

Thanks! Sorry again for the confusion. 🙏

Reply:
done. sorry for the delay. please check @holylovenia 🙏